command/server.go

```diff
@@ -1809,6 +1821,49 @@ func (c *ServerCommand) Run(args []string) int {
 				pFile.Close()
 			}
 
+			c.logger.Info(fmt.Sprintf("Wrote pprof files to: %s", pprofPath))
+		case <-heapProfCh:
+			path := c.flagHeapProfile
+			if _, err := os.Stat(path); err != nil {
+				c.logger.Error("Checking heap profile path failed", "error", err)
+				continue
+			}
+			pprofPath := filepath.Join(path, "vault-pprof")
+			err := os.MkdirAll(pprofPath, os.ModePerm)
+			if err != nil {
+				c.logger.Error("Could not create temporary directory for pprof", "error", err)
+				continue
+			}
+
+			dumps := []string{"goroutine", "heap", "allocs", "threadcreate", "profile"}
```

Comment:
The command line flag references "heap profile" but we're actually writing out many profiles here. Maybe it would be a good idea to rename the CLI flag to better reflect what we're actually doing?